utils/cache/cache.h

```diff
@@ -176,17 +177,48 @@ status_t cache_list_size(const char* const key, int* len);
 status_t cache_list_exist(const char* const key, const char* const value, const int value_len, bool* exist);
 
 /**
- * Pop an element from the list in in-memory cache
+ * @brief Pop an element from the list in in-memory cache
  *
- * @param[in] key Key for key-value starage
- * @param[out] res The element with assigning index
+ * @param key[in] Key for key-value storage
+ * @param res[out] The element with assigned index
  *
  * @return
  * - SC_OK on success
  * - non-zero on error
  */
 status_t cache_list_pop(const char* const key, char* res);
 
+/**
+ * @brief Get how many bytes redis has occupied.
+ *
+ * @return
+ * - Used size of the cache database on success
+ * - `-1` on error
+ */
+long int cache_occupied_space();
+
+/**
+ * @brief Initialize read/write lock for caching service
+ *
+ * @param rwlock[in] Pthread read/write lock object
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t cache_acquire(pthread_rwlock_t* rwlock);
```

Comment:
Should we operate on `pthread_rwlock_t *` object? Can we use more consistent ways to initialize?